controllers/azurecluster_reconciler.go

```diff
@@ -71,6 +72,7 @@ func newAzureClusterService(scope *scope.ClusterScope) (*azureClusterService, er
 			subnets.New(scope),
 			vnetpeerings.New(scope),
 			loadbalancers.New(scope),
+			privatelinks.New(scope),
```

Comment:
Just to make sure, we have these services in the order of dependency, i.e. VNet must be created before subnet. Is this the right spot for the private link service?